llrt_core/src/vm.rs

```diff
@@ -151,29 +149,33 @@ impl Resolver for BinaryResolver {
             return Ok(normalized_path.to_string());
         }
 
-        let path = self
-            .paths
-            .iter()
-            .find_map(|path| {
-                let path = path.join(normalized_path);
-                let bin_path = BinaryResolver::get_bin_path(&path);
-                if bin_path.exists() {
-                    return Some(bin_path);
-                }
-                get_js_path(path.to_str().unwrap())
-            })
-            .ok_or_else(|| Error::new_resolving(base, name))?;
+        let path = self.paths.iter().find_map(|path| {
+            let path = path.join(normalized_path);
+            let bin_path = CustomResolver::get_bin_path(&path);
+            if bin_path.exists() {
+                return Some(bin_path);
+            }
+            get_js_path(path.to_str().unwrap())
+        });
```

Comment:
This was originally thought to be the part of the BinaryResolver implementation that does not affect the resolution of mode_modules. (In other words, it cannot be resolved here, so it reaches the logic of node_modules path resolution.)

I didn't pursue this part too deeply, so I'll think about it some more by checking the contents of each variable while inserting `println!`

Reply:
By “node module resolution” you are referring to the resolution of submodules imported by relative paths within a Node.js package, rather than the resolution of the Node.js package itself? Since `get_js_path()` was the same functionality that we had been forced to implement in Loader, we removed the implementation we had added to Loader and used `get_js_path()`.

We believe that the means of resolving paths from Node.js package names must still be resolved using `get_module_path_in_node_modules()`.

Reply:
I cannot decide if CustomResolver::get_bin_path() is unnecessary or not. Here, I assume that for path files that do not contain an extension, if one with the extension “lrt” exists, it is returned.

Do I need to read bytecode-compiled files on the OS?